sdk/identity/azure_identity/src/process/mod.rs

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// cspell:ignore workdir
+
+use azure_core::{
+    credentials::AccessToken,
+    error::{Error, ErrorKind, Result},
+    process::Executor,
+};
+use std::{ffi::OsStr, marker::PhantomData, sync::Arc};
+
+use crate::env::Env;
+
+/// Runs a command in the appropriate platform shell and processes the output
+/// using the specified `OutputProcessor`.
+///
+/// - Windows: Runs `cmd /C {command}` in %SYSTEMROOT%
+/// - Everywhere else: Runs `/bin/sh -c {command}` in /bin
+pub(crate) async fn shell_exec<T: OutputProcessor>(
+    executor: Arc<dyn Executor>,
+    env: &Env,
+    command: &str,
+    _: PhantomData<T>,
+) -> Result<AccessToken> {
+    let (workdir, program, c_switch) = if cfg!(target_os = "windows") {
+        let system_root = env.var("SYSTEMROOT").map_err(|_| {
+            Error::message(
+                ErrorKind::Credential,
+                "SYSTEMROOT environment variable not set",
+            )
+        })?;
+        (system_root, "cmd", "/C")
+    } else {
+        ("/bin".to_string(), "/bin/sh", "-c")
+    };
+
+    let command_string = format!("cd {workdir} && {command}");
+    let args = vec![OsStr::new(c_switch), OsStr::new(&command_string)];
```

Comment:
You're going between `&str` to `String` and to `OsStr`, which is the same as `&str` on all but Windows. Instead, just declare your locals above as `&OsStr` and use `std::env::var_os`. For static strings, use `OsStr::new` directly.

Reply:
🤔 I could write `OsStr::new("literal")` above, but what's the benefit over doing that on this line? I agree the `String` is awkward but I don't see a way to avoid it because `env::var` returns a `String` and `env::var_os` returns an `OsString` i.e., still an owned string.

Reply:
Ultimately, it was trying to avoid the conversions back and forth - especially on Windows (UCS-2). You should be able to use `OsString`/`&Osstr` throughout instead of `String`/`&str` for efficiency.